.dotnet/examples/Chat/Example04_FunctionCallingStreaming.cs

```diff
@@ -12,6 +15,122 @@ public partial class ChatExamples
 {
     // See Example03_FunctionCalling.cs for the tool and function definitions.
 
+    #region
+    public class StreamingChatToolCallsBuilder
```

Comment:
One problem with chat streaming is that the tool calls are also streamed. In particular, the argument list is streamed. As you can imagine, there isn't really much that one can do with the individual chunks. Instead, users have to wait until the whole argument list has been streamed so that they can put it together and actually use it to call the function.

This `StreamingChatToolCallsBuilder` that I added in this example does the work of building the tool calls from the chunks. After GA, we should consider adding a helper like this to the actual library. If you look at the resulting example, this is super helpful.